maltoolbox/attackgraph/attackgraph.py

```diff
@@ -514,57 +524,70 @@ def _generate_graph(self) -> None:
+            for attack_step_name, attack_step in \
+                    lang_graph_asset.attack_steps.items():
                 logger.debug(
                     'Generating attack step node for %s.', attack_step_name
                 )
 
                 defense_status = None
                 existence_status = None
-                node_name = asset.name + ':' + attack_step_name
+                node_name = asset.name + ':' + attack_step.name
 
-                match (attack_step_attribs['type']):
+                match (attack_step.type):
                     case 'defense':
                         # Set the defense status for defenses
-                        defense_status = getattr(asset, attack_step_name)
+                        defense_status = getattr(asset, attack_step.name)
                         logger.debug(
-                            'Setting the defense status of %s to %s.',
+                            'Setting the defense status of \"%s\" to '
+                            '\"%s\".',
                             node_name, defense_status
                         )
 
                     case 'exist' | 'notExist':
-                        # Resolve step expression associated with (non-)existence
-                        # attack steps.
-                        (target_assets, attack_step) = _process_step_expression(
-                            self.lang_graph,
-                            self.model,
-                            [asset],
-                            attack_step_attribs['requires']['stepExpressions'][0])
-                        # If the step expression resolution yielded the target
-                        # assets then the required assets exist in the model.
-                        existence_status = target_assets != []
-
-                mitre_info = attack_step_attribs['meta']['mitre'] if 'mitre' in\
-                    attack_step_attribs['meta'] else None
+                        # Resolve step expression associated with
+                        # (non-)existence attack steps.
+                        existence_status = False
+                        for requirement in attack_step.get_all_requirements():
+                            target_assets = self._follow_expr_chain(
+                                    self.model,
+                                    [asset],
+                                    requirement
+                                )
+                            # If the step expression resolution yielded
+                            # the target assets then the required assets
+                            # exist in the model.
+                            if target_assets:
+                                existence_status = True
+                                break
+
+                    case _:
+                        pass
+
+                mitre_info = attack_step.info['mitre'] if\
+                    'mitre' in attack_step.info else None
```

Comment:
```suggestion
                mitre_info = attack_step.info.get('mitre', None)
```

Reply:
I think now that we have all of the info fields exposed via the language graph attack step we shouldn't handle the `mitre_info` separetely anymore. Do you agree?

Reply:
I think it makes sense.